packages/hoppscotch-backend/src/team-collection/team-collection.resolver.ts

```diff
@@ -141,7 +142,14 @@ export class TeamCollectionResolver {
     );
 
     if (E.isLeft(teamCollections)) throwErr(teamCollections.left);
-    return teamCollections.right;
+    return <TeamCollection>{
+      id: teamCollections.right.id,
+      title: teamCollections.right.title,
+      parentID: teamCollections.right.parentID,
+      data: !teamCollections.right.data
+        ? null
+        : JSON.stringify(teamCollections.right.data),
+    };
```

Comment:
We can move this type conversion in the service class (inside the `cast` function)

Reply:
We can't use the `cast` function inside this service method `getCollection` because `getCollection` is a helper function being used in a lot of locations in the `TeamCollection` module where its output it expected to be of type `DBTeamCollection` not `TeamCollection` hence, I moved it here to the resolver.